src/client/components/learnerGallery/index.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import CollectionPage from '../../containers/collection';
+import _ from 'lodash';
+
+class LearnerGallery extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      currentView: this.filterLearner(this.props.type),
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange (event) {
+    event.preventDefault();
+    this.setState({ currentView: event.target.value });
+  }
+
+  filterByName () {
+    const filteredLearner = this.filterLearner(this.props.type);
+    if (Array.isArray(this.state.currentView)) {
+      return filteredLearner;
+    }
+    return filteredLearner.filter(learner => {
+      if (learner.name.toLowerCase().startsWith(this.state.currentView.toLowerCase())) {
+        return learner;
+      }
+    });
+  }
+
+  filterLearner (type) {
```

Comment:
please refactor this function and split it into 2 functions - `filterByType` and `filterBySkill`